cmake/modules/dts.cmake

```diff
@@ -231,11 +230,11 @@ if(SUPPORTS_DTS)
   else()
     zephyr_file_copy(${ZEPHYR_DTS}.new ${ZEPHYR_DTS} ONLY_IF_DIFFERENT)
     zephyr_file_copy(${DEVICETREE_UNFIXED_H}.new ${DEVICETREE_UNFIXED_H} ONLY_IF_DIFFERENT)
-    zephyr_file_copy(${DEVICE_EXTERN_H}.new ${DEVICE_EXTERN_H})
-    file(REMOVE ${ZEPHYR_DTS}.new ${DEVICETREE_UNFIXED_H}.new ${DEVICE_EXTERN_H}.new)
+    file(WRITE ${DEVICE_EXTERN_H}
+"#error The contents of this file are now implemented directly in zephyr/device.h.")
```

Comment:
was there any case of direct `device_extern.h` inclusion? ie, was it a documented public header? Is it worth adding this?

Reply:
> was there any case of direct device_extern.h inclusion?

Not to my knowledge, but mcuboot used to include internal DT headers and the migration to the current generated macros broke the build there. I'm trying to be more helpful to users who are doing the wrong thing. Yes, they're wrong, but it's easy to point them in the right direction in this case.

>  ie, was it a documented public header?

No.

>  Is it worth adding this?

Yes, I think so, at **least** for 1 release, just as advice. Also consider that I just gave a talk about how the device extern mechanism works at ZDS that mentions this header explicitly. I think people who watched the talk will look for it and get confused without a little help here.